perses/annihilation/ncmc_switching.py

```diff
@@ -135,6 +136,29 @@ def _getAvailableParameters(self, system, prefix='lambda'):
                         parameters.append(parameter_name)
         return parameters
 
+    def _updateAlchemicalState(self, context, functions, value):
+        """
+        Update alchemical state using the specified lambda value.
+
+        Parameters
+        ----------
+        context : simtk.openmm.Context
+            The Context
+        functions : dict
+            A dictionary of functions
+        value : float
+            The alchemical lambda value
+
+        TODO: Improve function evaluation to better match Lepton and be more flexible in exact replacement of 'lambda' tokens
+
+        """
+        from parsing import NumericStringParser
```

Comment:
`pyparsing`: https://anaconda.org/anaconda/pyparsing

```
conda install --yes pyparsing
```
This is just a stop-gap measure. We'll need some way to evaluate the functions we feed to lepton to set the individual context parameters `lambda_*` based on `lambda` to 0 or 1 and the beginning and end of the protocol, unless I can think of a clever way to use a `CustomIntegrator` to do the calculation we need.